packages/ember-glimmer/lib/syntax/render.js

```diff
@@ -81,7 +83,18 @@ class AbstractRenderManager {
 class SingletonRenderManager extends AbstractRenderManager {
   create(definition, args, dynamicScope) {
     let { name, env } = definition;
-    let controller = env.owner.lookup(`controller:${name}`);
+    let controller = env.owner.lookup(`controller:${name}`) || generateControllerFactory(env.owner, name);
+
+    if (dynamicScope.outletState) {
```

Comment:
Does the non-singleton render (`{{render "foo" someModel}}`) not support this feature?